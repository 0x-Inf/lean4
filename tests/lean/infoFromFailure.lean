new_frontend

def A.foo {α : Type} [HasAdd α] (a : α) : α × α :=
(a, a + a)

def B.foo {α : Type} (a : α) : α × α :=
(a, a)

open A
open B

set_option trace.Meta.synthInstance true
set_option trace.Meta.isDefEq true
set_option trace.Elab true
-- `foo` is overloaded, the case `A.foo` is discarded because we don't have an instance `[HasAdd String]`.
-- However, we still want to see the trace since we used trace.Meta.synthInstance
#check foo "hello"

theorem ex : foo true = (true, true) :=
rfl
