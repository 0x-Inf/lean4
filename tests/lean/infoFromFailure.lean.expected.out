[Meta.synthInstance] HasAdd String ==> HasAdd String
[Meta.synthInstance] 
  [Meta.synthInstance] main goal HasAdd String
  [Meta.synthInstance.newSubgoal] HasAdd String
  [Meta.synthInstance.globalInstances] HasAdd String #[]
  [Meta.synthInstance] remaining fuel 999
  [Meta.synthInstance] failed
foo "hello" : String Ã— String
[Meta.synthInstance] HasAdd Bool ==> HasAdd Bool
[Meta.synthInstance] 
  [Meta.synthInstance] main goal HasAdd Bool
  [Meta.synthInstance.newSubgoal] HasAdd Bool
  [Meta.synthInstance.globalInstances] HasAdd Bool #[]
  [Meta.synthInstance] remaining fuel 999
  [Meta.synthInstance] failed
